Public/Src/Tools/Execution.Analyzer/Analyzers.Partner/ContentPlacementAnalyzer.cs

```diff
@@ -0,0 +1,532 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.ToolSupport;
+using BuildXL.Utilities;
+using BuildXL.Storage;
+using BuildXL.Pips.Operations;
+using BuildXL.Pips.Artifacts;
+using Newtonsoft.Json;
+using System.Diagnostics;
+using BuildXL.Cache.ContentStore.Interfaces.Tracing;
+
+// @cesar: I added this file
+namespace BuildXL.Execution.Analyzer
+{
+    internal partial class Args
+    {
+        public Analyzer InitializeContentPlacementAnalyzer()
+        {
+            int sampleSize = 0;
+            int sampleSizeHardLimit = int.MaxValue;
+            string buildQueue = null;
+            string buildId = null;
+            long buildStartTicks = 0;
+            double buildDurationMs = 0.0;
+            foreach (var opt in AnalyzerOptions)
+            {
+                if (opt.Name.StartsWith("sampleProportion", StringComparison.OrdinalIgnoreCase) ||
+                    opt.Name.StartsWith("sp", StringComparison.OrdinalIgnoreCase))
+                {
+                    sampleSize = ParseInt32Option(opt, 0, int.MaxValue);
+                }
+                if (opt.Name.StartsWith("sampleCountHardLimit", StringComparison.OrdinalIgnoreCase) ||
+                    opt.Name.StartsWith("schl", StringComparison.OrdinalIgnoreCase))
+                {
+                    sampleSizeHardLimit = ParseInt32Option(opt, 0, int.MaxValue);
+                }
+                if (opt.Name.StartsWith("buildQueue", StringComparison.OrdinalIgnoreCase) ||
+                    opt.Name.StartsWith("bq", StringComparison.OrdinalIgnoreCase))
+                {
+                    buildQueue = ParseStringOption(opt);
+                }
+                if (opt.Name.StartsWith("buildId", StringComparison.OrdinalIgnoreCase) ||
+                    opt.Name.StartsWith("bid", StringComparison.OrdinalIgnoreCase))
+                {
+                    buildId = ParseStringOption(opt);
+                }
+                if (opt.Name.StartsWith("buildStartTicks", StringComparison.OrdinalIgnoreCase) ||
+                    opt.Name.StartsWith("bst", StringComparison.OrdinalIgnoreCase))
+                {
+                    buildStartTicks = ParseInt64Option(opt, 0, int.MaxValue);
+                }
+                if (opt.Name.StartsWith("buildDurationMs", StringComparison.OrdinalIgnoreCase) ||
+                    opt.Name.StartsWith("bd", StringComparison.OrdinalIgnoreCase))
+                {
+                    buildDurationMs = ParseDoubleOption(opt, 0, double.MaxValue);
+                }
+
+            }

```

Comment:
nitpick: could you flip the newline to before the `return`?

Reply:
I've noticed in a  few places. Most of our code tries to follow the pattern of newline after '}' but not if there are two. If you could do a quick pass that would be appreciated. Thanks!

---
In reply to: [314074197](https://github.com/microsoft/BuildXL/pull/734#discussion_r314074197) [](ancestors = 314074197)